src/components/practiceTouch.tsx

```diff
@@ -50,16 +51,36 @@ export default function PracticeTouch({
       ))}
 
       <br />
-      <Button
-        variant="contained"
-        onClick={() => {
-          judgeAnswer(judge(brailleStrings, answer));
-        }}
-      >
-        答え合わせ
-      </Button>
-
-      {rightOrWrong}
-    </div>
+      {visible ? (
+        <Button
+          variant="contained"
+          onClick={() => {
+            judgeAnswer(
+              brailleDotCount === 6
+                ? judge(brailleStrings, answer)
+                : eightJudge(brailleStrings, answer)
+            );
+            setVisible(false);
+          }}
+        >
+          答え非表示
+        </Button>
+      ) : (
+        <Button
+          variant="contained"
+          onClick={() => {
+            judgeAnswer(
+              brailleDotCount === 6
+                ? judge(brailleStrings, answer)
+                : eightJudge(brailleStrings, answer)
+            );
+            setVisible(true);
+          }}
+        >
+          答え合わせ
+        </Button>
+      )}
```

Comment:
[IMO]
このように書くことも出来ます。

```suggestion
      <Button
        variant="contained"
        onClick={() => {
          judgeAnswer(
            brailleDotCount === 6
              ? judge(brailleStrings, answer)
              : eightJudge(brailleStrings, answer)
          );
          setVisible(!visible);
        }}
      >
        {visible ? "答え非表示" : "答え合わせ" }
      </Button>
```